staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -21,39 +21,246 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubectl/pkg/util/podutils"
+	"k8s.io/utils/pointer"
 )
 
-// profileLegacy represents the legacy debugging profile which is backwards-compatible with 1.23 behavior.
-func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object) error {
+type legacyProfile struct {
+}
+
+type generalProfile struct {
+}
+
+type baselineProfile struct {
+}
+
+type restrictedProfile struct {
+}
+
+func (p *legacyProfile) Apply(pod *corev1.Pod, containerName string, target runtime.Object) error {
 	switch target.(type) {
 	case *corev1.Pod:
 		// do nothing to the copied pod
 		return nil
 	case *corev1.Node:
-		const volumeName = "host-root"
-		pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
-			Name: volumeName,
-			VolumeSource: corev1.VolumeSource{
-				HostPath: &corev1.HostPathVolumeSource{Path: "/"},
-			},
-		})
+		MountRootPartition(pod, containerName)
+		UseHostNamespaces(pod)
+		return nil
+	default:
+		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
+	}
+}
+
+type debugStyle int
```

Comment:
nit: It would be better to move constants at the beginning of this file.

Reply:
good idea - done in 7f7ef70651b9d81cf479647805019e0aec3e484e